diff/order.go

```diff
@@ -0,0 +1,84 @@
+package diff
+
+import "github.com/kong/deck/types"
+
+/*
+                                       Root
+                                         |
+         +----------+----------+---------+------------+---------------+
+         |          |          |         |            |               |
+         v          v          v         v            v               v
+L1    Service    RbacRole  Upstream  Certificate  CACertificate  Consumer ---+
+      Package        |         |        |     |      |                |      |
+        |            v         v        v     |      v                v      |
+L2      |        RBACRole   Target     SNI    +-> Service       Credentials  |
+        |        Endpoint                         |  |              (7)      |
+        |                                         |  |                       |
+        |                                         |  |                       |
+L3      +---------------------------> Service <---+  +-> Route               |
+        |                             Version        |     |                 |
+        |                                 |          |     |                 |
+        |                                 |          |     v                 |
+L4      +----------> Document   <---------+          +-> Plugins  <----------+
+*/
+
+var dependencyOrder = [][]string{
```

Comment:
> Creating deep copies and implementing reversal just to make it possible to iterate over an array in a reverse order seems superfluous. Consider the following alternative that means less code (and less memory copying):

Although this is more code, I want to keep the ordering separate. The motivation here is that the ordering will be injected into the Syncer rather than hard-coding it in there.

Reply:
I took your suggestion and reworded it a little bit.